cmd/dep/root_analyzer.go

```diff
@@ -53,6 +57,71 @@ func (a *rootAnalyzer) InitializeRootManifestAndLock(dir string, pr gps.ProjectR
 	return
 }
 
+func (a *rootAnalyzer) cacheDeps(pr gps.ProjectRoot, pkgT pkgtree.PackageTree) {
+	packages := make(map[string]bool)
+	dependencies := make(map[gps.ProjectRoot][]string)
+	logger := a.ctx.Err
+
+	syncDep := func(pr gps.ProjectRoot, sm gps.SourceManager) {
+		message := fmt.Sprintf("Cached %s", pr)
+		if err := sm.SyncSourceFor(gps.ProjectIdentifier{ProjectRoot: pr}); err != nil {
+			message = fmt.Sprintf("Unable to cache %s - %s", pr, err)
+		}
+		logger.Printf(message)
+	}
+
+	for _, v := range pkgT.Packages {
```

Comment:
I think that you can use`a.directDeps` instead of pkgT, which is the set of repositories represented in by the import paths in `pkgT`. Then we don't need to pass in `pkgT` or deal with parsing project roots, or skipping duplicates.